src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -0,0 +1,96 @@
+package seedu.address.logic.commands;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+
+/**
+ * Allows users to view full profile of their friends at the right panel.
+ */
+public class ViewCommand extends Command {
+
+    public static final String COMMAND_WORD = "view";
+    public static final String USAGE = "view <index> : Allows you to view the profile of the person who"
+            + "has the index\n"
+            + "view n/<name> : Allows you to view profile for the specific person"
+            + "view : Shows your own profile instead";
+    private String name;
+    private Index index;
+
+    /**
+     * Creates a View Command with the person's name.
+     * @param name
+     */
+    public ViewCommand(String name) {
+        this.name = name;
+        this.index = null;
+    }
+
+    /**
+     * Creates a View Command.
+     */
+    public ViewCommand() {
+        this.name = null;
+        this.index = null;
+    }
+
+    /**
+     * Creates a View Command with the person's index.
+     * @param index
+     */
+    public ViewCommand(Index index) {
+        this.name = null;
+        this.index = index;
+    }
```

Comment:
Hmm a little strange to have 3 constructors here but I guess should be fine.

Reply:
Yes, see latest PR. Thanks for the suggestion. I wanted to keep it simple first and slowly incorporate Optional. It has been done. See latest commit [https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/82/commits/3ac80cffe300d0a00804d56cdaff389ecddd679d](url)